ItemList/Tests/ItemList.DependencyInjection.Tests/ContratsImplementationTests.cs

```diff
@@ -19,7 +19,65 @@ internal class ContratsImplementationTests
         [SetUp]
         public void SetUp()
         {
-            _containerMock = new ContainerMock();
+            _containerMock = new ContainerMock(new HashSet<string>());
+            SetUpResolverMock();
+        }
+
+        [Test]
+        public void AllContractsHaveRegisteredImplementation()
+        {
+            var bootstrap = new Bootstrap(BootstrapFactory.Bootstrappers.Value, _resolverMock);
+            var expectedContracts = GetExpectedContracts();
+
+            bootstrap.RegisterDependencies();
+            var actualContracts = _containerMock.RegisteredContracts.ToArray();
+
+            Assert.That(
+                actualContracts,
+                Is.EquivalentTo(expectedContracts),
+                GetErrorMessage(actualContracts, expectedContracts)
+                );
+        }
+
+        private static string GetErrorMessage(ICollection<string> actualContracts, ICollection<string> expectedContracts)
```

Comment:
You spent quite some time on formatting the message. I appreciate your effort, yet these are tests for you (and other developers of course) and most of us do not see well through pink UX glasses. The code is quite complex, from my point of view and does not cope well with different sizes of tests results window, however, if you see benefit in it (or at least you feel you really put quite some effort to it as I do feel), do not change it (since it is your code, under your production :wink:)